docs/template_plugin/tests/functional/op_reference/roi_pooling.cpp

```diff
@@ -0,0 +1,130 @@
+#include <ngraph/ngraph.hpp>
+#include <tuple>
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+struct ROIPoolingParams {
+    ROIPoolingParams(const size_t iH, const size_t iW, const size_t ch, const size_t rois,
+                     const size_t oH, const size_t oW, const float sS, const std::string mode,
+                     void inputInit(float&),
+                     const std::vector<float>& proposalValues, const std::vector<float>& outputValues)
+        : inputH(iH), inputW(iW), channelCount(ch), roiCount(rois), outputH(oH), outputW(oW), spatialScale(sS), poolingMode(mode),
+        proposal(CreateBlob(element::f32, proposalValues)), refData(CreateBlob(element::f32, outputValues)) {
+        auto inputSize = iH * iW * ch;
+        std::vector<float> inputValues(inputSize);
+        std::for_each(inputValues.begin(), inputValues.end(), inputInit);
+        featureMap = CreateBlob(element::f32, inputValues, inputSize);
+    }
+    size_t inputH;
+    size_t inputW;
+    size_t channelCount;
+    size_t roiCount;
+    size_t outputH;
+    size_t outputW;
+    float spatialScale;
+    std::string poolingMode;
+    InferenceEngine::Blob::Ptr featureMap;
+    InferenceEngine::Blob::Ptr proposal;
+    InferenceEngine::Blob::Ptr refData;
+};
+
+class ReferenceRoiPoolingLayerTest : public testing::TestWithParam<ROIPoolingParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params.inputH, params.inputW, params.channelCount, params.roiCount,
+                                  params.outputH, params.outputW, params.spatialScale, params.poolingMode);
+        inputData = {params.featureMap, params.proposal};
+        refOutData = {params.refData};
+    }
+    static std::string getTestCaseName(const testing::TestParamInfo<ROIPoolingParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "IS=" << param.inputH << "," << param.inputW << "_";
+        result << "OS=" << param.outputH << "," << param.outputW << "_";
+        result << "Ch=" << param.channelCount << "_";
+        result << "Rois=" << param.roiCount << "_";
+        result << "Ss=" << param.spatialScale << "_";
+        result << "Mode=" << param.poolingMode << "_";
+        result << std::to_string(obj.index);
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const size_t i_h, const size_t i_w, const size_t ch, const size_t roi_count,
+                                                    const size_t o_h, const size_t o_w, const float spat_scale, const std::string mode) {
+        Shape feat_map_shape{1, ch, i_h, i_w};
+        Shape rois_shape{roi_count, 5};
+        Shape pooled_shape{o_h, o_w};
+        Shape output_shape{roi_count, ch, o_h, o_w};
+
+        const auto feat_map = std::make_shared<op::Parameter>(element::f32, feat_map_shape);
```

Comment:
I thought that ROIPooling supports not only f32 data type. Please extend these tests for all supported data types

Reply:
Test extended for supported types (floating point)